examples/gcp_http/example_http_function.py

```diff
@@ -0,0 +1,18 @@
+import hashlib
+
+import flask
+import requests
+
+
+def handler(request: flask.Request):
```

Comment:
This is too fancy for Python 2.7. You'll have to use:
```python
def handler(request):
    # type: (flask.Request) -> Dict[str, str]
```
Or go without typing ascriptions.

Reply:
Yeah,  ended up just removing. Now getting a similar error as local. I'm guessing Flask or one of its dependencies that doesn't have an appropriate wheel or something?

Reply:
Aha, no. This is a problem with old Pex not presenting a fully-working resolver (it's bespoke in old Pex). So, it simply can't handle the double-requirement with env markers to pick between the two technique. I just tried locally and this works fine for both pre-1.6 and post-1.6 tests:
```
$ cat examples/gcp_http/requirements.txt
flask<2
requests
```

Reply:
:/ CI still doesn't like me: https://github.com/pantsbuild/lambdex/pull/22/checks?check_run_id=3457507844

Reply:
Hrm. Yeah. pre-1.6 works for py39 but not py27. I repro that locally. Looking...

Reply:
Ok. You will not be able to get pre-1.6 Pex working with Flask and Python 2.7. The old be-spoke resolved does not look at `Requires-Python` package metadata, so it grabs MarkupSafe 2+ which have ~: `Requires-Python: >=3.6`. This set of problems cascades.  As such anyone using old Pex is very unlikely to be building new PEX files with Flask dependencies because its so un-workable. These folks will have to use new Pex if they also want to test lambdex for gcp_http - but since this is a new feature anyhow - that does not feel onerous at all.

It's probably easiest to change the integration tests commands into a bash or python script and then use conditional logic to de-activate the gcp_http test for pre-1.6 Python 2.7. It might be even better to just draw a line in the sand and say pre-1.6 gets no new feature support and seal the integration tests commands for pre-1.6 to the prior list and have post-1.6 use that list plus the new gcp_http commands.

If that doesn't make sense, speak up.

Thanks for hanging in there.

Reply:
Makes sense to me, although I had pre-1.6 tests failing on python 3.9 as well 🤔

But if we're removing support for pre-1.6 altogether, it all should be good. I'll make the new integration tests and add some docs to that effect. 

Thanks for your help on this!

Reply:
To be clear, I'm just hoping to remove (never add) support for gcp-http in pre-1.6 Pex & Lambdex in this PR. Fully removing support for pre-1.6 Pex is for another day and a dedicated PR / release, etc.

Reply:
Done. Everything is passing!

Finally!